File: leetcode/easy/3120-count-the-number-of-special-characters-i/3120-count-the-number-of-special-characters-i.py
# // 3110-score-of-a-string
# // URL: https://leetcode.com/problems/score-of-a-string/description/

class Solution:
    def numberOfSpecialChars(self, word: str) -> int:
        lowerSet = set()
        upperSet = set()
        count = 0

        for i in range(len(word)):
            if word[i].islower():
                lowerSet.add(word[i])
            else:
                upperSet.add(word[i])

        for element in lowerSet:
            if element.upper() in upperSet:
                count += 1

        return count

File: leetcode/easy/3120-count-the-number-of-special-characters-i/3120-count-the-number-of-special-characters-i.kt
// 3105-longest-strictly-increasing-or-strictly-decreasing-subarray
// URL: https://leetcode.com/problems/longest-strictly-increasing-or-strictly-decreasing-subarray/description/


class Solution {
    fun numberOfSpecialChars(word: String): Int {
        // I used this because hashsetOf's lookup worst time complexity is O(n)
        var lowerMutableSet = mutableSetOf<Char>()
        var upperMutableSet = mutableSetOf<Char>()

        for (char in word) {
            if (char.isLowerCase()) {
                lowerMutableSet.add(char)
            } else {
                upperMutableSet.add(char)
            }
        }

        var count = 0

        for (char in lowerMutableSet) {
            if (upperMutableSet.contains(char.uppercaseChar())) {
                count++
            }
        }

        return count
    }
}
